pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -18,20 +19,42 @@ const MinSize = 16 // bytes
 // Fingerprint is used to identify a file
 // A file's fingerprint is the first N bytes of the file
 type Fingerprint struct {
-	FirstBytes []byte
+	FirstBytes   []byte
+	HashBytes    uint64
+	BytesUsed    int
```

Comment:
Is this always intended to reflect `len(FirstBytes)`? If so, is there really a performance benefit to maintaining it separately?